lib/setup.js

```diff
@@ -2204,12 +2204,13 @@ function dbConnect(onlyCheck, params, callback) {
         params    = null;
     }
     params = params || {};
+
+    const config = JSON.parse(fs.readFileSync(tools.getConfigFileName(), 'utf8'));
```

Comment:
I read this exact line so often. Shouldn't that be a utility function somewhere? E.g. in `tools`

Reply:
With my change in "dbConnect" to return config in the callback as last parameter all other "setup" places could be changed later to use this instead of including by themself. And then only this one place is left ;-)